sockets.go

```diff
@@ -157,6 +157,49 @@ type JSONConnection struct {
 	Receiver reflect.Value
 }
 
+// Binary Connection connects a websocket message connection to a []byte
+// channel.
+type BinaryConnection struct {
+	*Connection
+
+	// Sender is the []byte channel used for sending out binary data to the client.
+	// This channel gets mapped for the next handler to use and is asynchronous
+	// unless the SendChannelBuffer is set to 0.
+	Sender chan []byte
+
+	// Receiver is the []byte channel used for receiving binary data from the client.
+	// This channel gets mapped for the next handler to use and is asynchronous
+	// unless the RecvChannelBuffer is set to 0.
+	Receiver chan []byte
+}
+
+// Messages returns a websocket handling middleware. It can only be used
+// in handlers for HTTP GET.
+// IMPORTANT: The last handler in your handler chain must block in order for the
+// connection to be kept alive.
+// It maps four channels for you to use in the follow-up Handler(s):
+// - A receiving string channel (<-chan string) on which you will
+//   receive all incoming strings from the client
+// - A sending string channel (chan<- string) on which you will be
+//   able to send strings to the client.
+// - A receiving error channel (<-chan error) on which you will receive
+//   errors occurring while sending & receiving
+// - A receiving disconnect channel  (<-chan bool) on which you will receive
+//   a message only if the connection is about to be closed following an
+//   error or a client disconnect.
+// - A sending done channel  (chan<- bool) on which you can send as soon as you wish
+//   to disconnect the connection.
+// The middleware handles the following for you:
+// - Checking the request for cross origin access
+// - Doing the websocket handshake
+// - Setting sensible options for the Gorilla websocket connection
+// - Starting and terminating the necessary goroutines
+// An optional sockets.Options object can be passed to Messages to overwrite
+// default options mentioned in the documentation of the Options object.
```

Comment:
This documentation needs updating so it reflects that this is about `[]byte` channels